tower-posix-tests/tower-posix-tests.cabal

```diff
@@ -1,4 +1,4 @@
-name:                ivory-tower-posix
+name:                tower-posix-tests
```

Comment:
I copied the structure from the stm32 backend 
As an example : 
ivory-tower-stm32/tower-freertos-stm32-tests

It is possible to change it, but it should be done ideally for both backend.